cluster-autoscaler/processors/customresources/gpu_processor.go

```diff
@@ -42,13 +42,25 @@ func (p *GpuCustomResourcesProcessor) FilterOutNodesWithUnreadyResources(context
 	newReadyNodes := make([]*apiv1.Node, 0)
 	nodesWithUnreadyGpu := make(map[string]*apiv1.Node)
 	for _, node := range readyNodes {
-		_, hasGpuLabel := node.Labels[context.CloudProvider.GPULabel()]
-		gpuAllocatable, hasGpuAllocatable := node.Status.Allocatable[gpu.ResourceNvidiaGPU]
-		directXAllocatable, hasDirectXAllocatable := node.Status.Allocatable[gpu.ResourceDirectX]
```

Comment:
I think the change breaks the DirectX part, right? This is the only place handling DirectX, and it predates the addition of `GetNodeGpuConfig()`. I don't think we've taken it into account when implementing `GetNodeGpuConfig()`, so we can't just migrate to it. We should probably call `GetNodeGpuTarget()` just to determine if the GPU is exposed via DRA or device plugin, then leave the current logic hardcoding both NvidiaGpu and DirectX. WDYT?